Code/Framework/AzFramework/AzFramework/DocumentPropertyEditor/AdapterBuilder.h

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#pragma once
+
+#include <AzCore/std/containers/stack.h>
+#include <AzFramework/DocumentPropertyEditor/DocumentAdapter.h>
+#include <AzFramework/DocumentPropertyEditor/DocumentSchema.h>
+#include <AzFramework/DocumentPropertyEditor/PropertyEditorNodes.h>
+
+namespace AZ::DocumentPropertyEditor
+{
+    //! Helper class that builds a DOM suitable for usage in a DocumentAdapter.
+    //! Uses a visitor pattern to establish node elements.
+    class AdapterBuilder
+    {
+    public:
+        //! Creates an AdapterBuilder.
+        //! A routing adapter may optionally be specified to enable routing via calls to
+        //! NestedAdapter.
+        //! Constructing an AdapterBuilder immediately opens an Adapter tag.
```

Comment:
What's an adapter tag? It's not obvious from this PR.

Reply:
That meant `BeginAdapter` but I actually stopped doing that - comment removed entirely